src/core/thread.cpp

```diff
@@ -15,14 +16,16 @@ Thread::Thread() : native_handle{0} {}
 
 Thread::~Thread() {}
 
-int Thread::Create(ThreadFunc func, void* arg) {
+int Thread::Create(ThreadFunc func, void* arg, const ::Libraries::Kernel::PthreadAttr* attr) {
 #ifdef _WIN64
     native_handle = CreateThread(nullptr, 0, (LPTHREAD_START_ROUTINE)func, arg, 0, nullptr);
     return native_handle ? 0 : -1;
 #else
     pthread_t* pthr = reinterpret_cast<pthread_t*>(&native_handle);
     pthread_attr_t pattr;
     pthread_attr_init(&pattr);
+    pthread_attr_setstack(&pattr, attr->stackaddr_attr, attr->stacksize_attr);
+    pthread_attr_setguardsize(&pattr, attr->guardsize_attr);

```

Comment:
According to manual guardsize is ignoring when setting user allocated stacks so this is probably not needed